include/os/linux/spl/sys/cred.h

```diff
@@ -45,32 +45,73 @@ typedef struct cred cred_t;
 #define	SGID_TO_KGID(x)		(KGIDT_INIT(x))
 #define	KGIDP_TO_SGIDP(x)	(&(x)->val)
 
-static inline uid_t zfs_uid_into_mnt(struct user_namespace *mnt_ns, uid_t uid)
+extern boolean_t zfs_is_init_userns(struct user_namespace *ns);
+extern struct user_namespace *zfs_get_init_userns(void);
+
+static inline struct user_namespace *zfs_i_user_ns(struct inode *inode)
+{
+#ifdef HAVE_SUPER_USER_NS
```

Comment:
Do you support kernels without sb->s_user_ns?

Reply:
Yes we do.